dialects/linux/dsock.c

```diff
@@ -3633,7 +3633,7 @@ net6a2in6(as, ad)
 	    buf[8] = '\0';
 	    ep = (char *)NULL;
 	    if ((ad->s6_addr32[i] = (uint32_t)strtoul(buf, &ep, 16))
-	    ==  (uint32_t)UINT32_MAX || !ep || *ep)
+	    >  (uint32_t)UINT32_MAX || !ep || *ep)
```

Comment:
In my understanding, the value (V) of the left side of `>` operator is never greater than (uint32_t)UINT32_MAX.